utils/find_actors.py

```diff
@@ -0,0 +1,72 @@
+import argparse
+import ast
+import os
+
+
+def is_direct_actor_def(ast_node):
+    if not isinstance(ast_node, ast.ClassDef):
+        return False
+
+    has_actor_base = False
+    for class_base in ast_node.bases:
+        # We are looking for direct name 'Actor'
+        if not isinstance(class_base, ast.Name):
+            continue
+
+        if class_base.id == 'Actor':
+            has_actor_base = True
+            break
+
+    return has_actor_base
+
+
+def extract_actor_name_from_def(actor_class_def):
+    actor_name = None
+    for child in actor_class_def.body:
```

Comment:
Honestly, if I see more than one `for` in an expression it becomes hard for me to read it, maybe because I am not not used to seeing these in the wild. But I went ahead and I pulled the 'filtering for assignments' into a named generator so the loop body is simpler while not making maintainers to read nested inline for loops.